### PR TITLE
Fix some build warnings

### PR DESCRIPTION
**Pull request by remicollet**
As I think tons of warnings (I'm using -Wall option) create lot of noise and can hide real problems, this is a first set of fix for some build warnings.

P.S. I will try to work on some others later, when I will have some spare time (and after getting some feedback)

### CODE REVIEW

**Comment by Hadis-Fard** [comment]
@remicollet thanks a ton for your contribution, I'll start reviewing the changes right away.

**Comment by Hadis-Fard** [comment]
@remicollet finished the review, super appreciate the changes you have sent.

File: source/sqlsrv/conn.cpp [hunk 1]
```diff
@@ -1120,7 +1120,7 @@ void sqlsrv_conn_close_stmts( ss_sqlsrv_conn* conn TSRMLS_DC )
 
 		try {
 			// this would call the destructor on the statement.
-			int zr = zend_list_close(Z_RES_P(rsrc_ptr));
```
Comment (remicollet) [inline source/sqlsrv/conn.cpp]: Indeed, sqlsrv_stmt_dtor or ~ss_sqlsrv_stmt doesn't raise any exception, so try/catch seems unneeded.

Comment (remicollet) [inline source/sqlsrv/conn.cpp]: done

**Comment by Hadis-Fard** [comment]
@remicollet Accepted the PR, thanks 👍

**Pull request merged**